task/deprecated-image-check/0.1/deprecated-image-check.yaml

```diff
@@ -33,62 +35,57 @@ spec:
-        http_code=$(curl -s -k -o $(workspaces.sanity-ws.path)/repository_data.json -w '%{http_code}' "https://catalog.redhat.com/api/containers/v1/repositories/registry/$(params.IMAGE_REGISTRY)/repository/$(params.IMAGE_REPOSITORY)")
+        IFS=$'\n' read -d'' -r -a IMAGE_ARRAY < <(echo -n $BASE_IMAGES_DIGESTS | sed 's/\\n/\'$'\n''/g' && printf '\0')
+        for BASE_IMAGE in ${IMAGE_ARRAY[@]};
+        do
+            IFS=:'/' read -r IMAGE_REGISTRY IMAGE_WITH_TAG <<< "$BASE_IMAGE"; echo "[$IMAGE_REGISTRY] [$IMAGE_WITH_TAG]"
+            IMAGE_REPOSITORY=`echo $IMAGE_WITH_TAG | cut -d ":" -f1`
+            IMAGE_REGISTRY=${IMAGE_REGISTRY//registry.redhat.io/registry.access.redhat.com}
 
-        echo "Response code: $http_code"
-        echo -n $http_code > $(results.PYXIS_HTTP_CODE.path)
+            echo "Querying Pyxis for $(BASE_IMAGE)..."
+            http_code=$(curl -s -k -o $(workspaces.sanity-ws.path)/${IMAGE_REPOSITORY}/repository_data.json -w '%{http_code}' "https://catalog.redhat.com/api/containers/v1/repositories/registry/$(IMAGE_REGISTRY)/repository/$(IMAGE_REPOSITORY)")
+            echo "Response code: $http_code"
+            echo -n $http_code $IMAGE_REGISTRY $IMAGE_REPOSITORY \n>> $(results.PYXIS_HTTP_CODE.path)
+        done
 
     # Run the tests and save output
     - name: run-conftest
       image: quay.io/redhat-appstudio/hacbs-test:latest
       script: |
         #!/usr/bin/env sh
-        http_code=$(cat $(results.PYXIS_HTTP_CODE.path))
-
-        if [ "$http_code" == "200" ];
-        then
-          echo "Running conftest using $(params.POLICY_DIR) policy, $(params.POLICY_NAMESPACE) namespace"
-          /usr/bin/conftest test --no-fail $(workspaces.sanity-ws.path)/repository_data.json \
-          --policy $(params.POLICY_DIR) --namespace $(params.POLICY_NAMESPACE) \
-          --output=json 2> $(workspaces.sanity-ws.path)/stderr.txt | tee $(workspaces.sanity-ws.path)/deprecated_image_check_output.json
-          echo "Done!"
-          exit 0
-        elif [ "$http_code" == "404" ];
-        then
-          echo "Registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) not found in Pyxis" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        else
-          echo "Unexpected error (HTTP code $http_code) occured for registry/image $(params.IMAGE_REGISTRY)/$(params.IMAGE_REPOSITORY) during running conftest" > $(workspaces.sanity-ws.path)/stderr.txt
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
-
-    # Format the test results and save them as the HACBS_TEST_OUTPUT Tekton result
-    - name: test-format-result
-      image: quay.io/redhat-appstudio/hacbs-test:latest
-      script: |
-        if [ ! -z $(cat $(workspaces.sanity-ws.path)/stderr.txt) ]
-        then
-          echo "The deprecated-image-check test meets the following error:"
-          cat $(workspaces.sanity-ws.path)/stderr.txt
-        fi
-
+        counter=0
```

Comment:
I think it might be good to name the counter to be a little more descriptive, like `failures_counter` or similar

Reply:
done